### PR TITLE
Client/server correct profiles

### PR DESCRIPTION
**Pull request by Mathixx**
Corrected the APIs of https://spec.matrix.org/v1.11/client-server-api/#profiles

Added TO DOs to complete when server-server API will be available
Added TO DOs following synapses implementation of matrix Protocol

Test coverage is deliberately incomplete for getProfiles related API to make sure it is not forgotten when server-server is available.

### CODE REVIEW

File: packages/matrix-client-server/src/profiles/getProfiles.ts [hunk 1]
```diff
@@ -63,45 +66,44 @@ export const getAvatarUrl = (
...
-        })
+          })
+          .catch((e) => {
+            /* istanbul ignore next */
+            send(res, 500, errMsg('unknown', e))
+            /* istanbul ignore next */
+            clientServer.logger.error('Error querying profiles:', e)
+          })
+      } else {
+        // TO DO : Have a look on remote server via federation
```
Comment (guimard) [inline packages/matrix-client-server/src/profiles/getProfiles.ts:95]: TODO

File: packages/matrix-client-server/src/profiles/getProfiles.ts [hunk 2]
```diff
@@ -111,38 +113,38 @@ export const getDisplayname = (
...
+        send(res, 500, errMsg('unknown', 'Cannot get profile of a remote user'))
```
Comment (guimard) [inline packages/matrix-client-server/src/profiles/getProfiles.ts:143]: add return (security)

File: packages/matrix-client-server/src/profiles/changeProfiles.ts [hunk 3]
```diff
@@ -50,30 +137,101 @@ export const changeAvatarUrl = (
...
+        console.log('got admin')
+        byAdmin = response[0].admin as number
+      } catch (e) {
+        /* istanbul ignore next */
+        clientServer.logger.error('Error checking admin:', e)
+        /* istanbul ignore next */
+        send(res, 500, errMsg('unknown', 'Error checking admin'))
+      }
+
+      console.log('finished checking admin')
```
[the remark this answers is not in the document]
Comment (Mathixx) [reply packages/matrix-client-server/src/profiles/changeProfiles.ts:181]: You're absolutely right, I forgot to delete my console.logs, they were there for testing purposes only, I removed them all !

File: packages/matrix-client-server/src/profiles/getProfiles.ts (hunk 2 above)
Comment (Mathixx) [reply packages/matrix-client-server/src/profiles/getProfiles.ts:143]: when adding a return, I have a "unecessary return statement" eslint error
Do I still add it ?

**Pull request merged**